src/gt4py/next/program_processors/runners/gtfn.py

```diff
@@ -46,15 +46,32 @@ def convert_arg(arg: Any) -> Any:
         return arg
 
 
+def iter_size_args(args: tuple[Any, ...]) -> Iterator[int]:
+    for arg in args:
+        match arg:
+            case tuple():
+                yield from iter_size_args((arg[0],))
+            case common.Field():
+                yield from arg.ndarray.shape
+            case _:
+                pass
+
+
 def convert_args(
-    inp: stages.CompiledProgram, device: core_defs.DeviceType = core_defs.DeviceType.CPU
+    inp: stages.ExtendedCompiledProgram, device: core_defs.DeviceType = core_defs.DeviceType.CPU
 ) -> stages.CompiledProgram:
     def decorated_program(
-        *args: Any, offset_provider: dict[str, common.Connectivity | common.Dimension]
+        *args: Any,
+        offset_provider: dict[str, common.Connectivity | common.Dimension],
+        out: Any = None,
     ) -> None:
+        if out is not None:
+            args = (*args, out)
         converted_args = [convert_arg(arg) for arg in args]
         conn_args = extract_connectivity_args(offset_provider, device)
-        return inp(*converted_args, *conn_args)
+        return inp(
+            *converted_args, *(iter_size_args(args) if inp.implicit_domain else ()), *conn_args
```

Comment:
This is where the the `implicit_domain` flag is consumed and size arguments are conditionally generated in the GTFN backend.